src/environmentd/src/bin/environmentd/main.rs

```diff
@@ -557,6 +558,10 @@ pub struct Args {
     #[clap(long, env = "AWS_ACCOUNT_ID")]
     aws_account_id: Option<String>,
 
+    /// The Materialize role arn to assume when connecting to external user's AWS account.
+    #[clap(long, env = "AWS_EXTERNAL_CONNECTION_ROLE")]
+    aws_external_connection_role: Option<String>,
```

Comment:
@jubrad Using `AWS_EXTERNAL_CONNECTION_ROLE` to pass the new role to assume. Lmk if you would prefer something else.